joblib/test/test_parallel.py

```diff
@@ -974,3 +991,23 @@ def test_backend_batch_statistics_reset(backend):
 
     # Tolerance in the timing comparison to avoid random failures on CIs
     assert test_time / ref_time <= 1 + relative_tolerance
+
+
+@with_numpy
+@parametrize('backend', PARALLEL_BACKENDS)
+@parametrize('n_jobs', [1, 2, -2, -1])
+def test_deadlock_with_generator(backend, n_jobs):
+    # pytest-timeout is set up so this will fail if it falls in a deadlock
+
+    def func(arg):
+        result = np.ones(int(5*1e5), dtype=bool)
```

Comment:
Why do you need to use numpy in this test? Given that numpy is an optional dependency of joblib, we try to minimize its use.

That said, I do not fully understand the purpose of this test. Maybe a small comment at the beginning of the test for why it's useful.

Reply:
This example used by @lesteve used to lead to a deadlock with `multiprocessing` backend. Numpy is not mandatory, any object that is big enough so that the pickle takes some time should do.